### PR TITLE
✨ Add VirtualMachineSpec.InstanceUUID

### PR DESCRIPTION
**Pull request by dougm**
For VM Import, we want the desired instance UUID to match the underlying instance UUID of the VM

Note: needs #492 merged before doc update

### CODE REVIEW

**Comment by akutz** [comment]
Thanks for this @dougm. One quick note, PTAL at https://github.com/vmware-tanzu/vm-operator/pull/518 and mimic what was done there for this new field. Specifically the protection of the field. Thanks!

File: Makefile
Comment (akutz) [inline Makefile:265]: Why this change?

File: api/v1alpha3/virtualmachine_types.go
Comment (akutz) [inline api/v1alpha3/virtualmachine_types.go:455]: Please leave an empty line between the field and the annotations for the next field.

File: api/v1alpha3/virtualmachine_types.go
Comment (akutz) [inline api/v1alpha3/virtualmachine_types.go:454]: I would also add an additional comment indicating this is *not* the same as any potential Cloud-Init instance ID, briefly describing the difference between the two fields.

Comment (dougm) [reply api/v1alpha3/virtualmachine_types.go:458]: Done

**Comment by dougm** [comment]
> PTAL at https://github.com/vmware-tanzu/vm-operator/pull/518 and mimic what was done there for this new field. Specifically the protection of the field

@akutz Sure, I'll wait for #518 to be merged, then add protection for InstanceUUID before creating more merge conflicts

File: api/v1alpha1/virtualmachine_conversion_test.go
Comment (dougm) [reply api/v1alpha1/virtualmachine_conversion_test.go]: Changed to NewString

**Comment by dougm** [comment]
> PTAL at https://github.com/vmware-tanzu/vm-operator/pull/518 and mimic what was done there for this new field. Specifically the protection of the field

@akutz rebased + added this

I still need to update docs. Need to merge #492 first, or I can just fold that into this PR if you prefer

**Review by bryanv — APPROVED** [review]

File: Makefile
Comment (akutz) [reply Makefile:265]: So this was addressed just now in #523 . You can undo your change @dougm . Apparently between Go 1.21 and Go 1.22, cross-module commands were restricted. From a Slack conversation with @dilyar85:

> Hi Andrew. I noticed that running `go generate ./api/...` in VMOP returns the following error for Go 1.22 (the previous version 1.21 doesn’t have this issue):
> 
> ```shell
> $ go version
> go version go1.22.3 darwin/amd64
> 
> $ go generate ./api/...
> pattern ./api/...: main module (github.com/vmware-tanzu/vm-operator) does not contain package > github.com/vmware-tanzu/vm-operator/api
> ```
> 
> I checked the [Go 1.22 Release Notes](https://tip.golang.org/doc/go1.22) but didn’t see anything related. Also did some searching online and found no relevant results.
> Since we use Go 1.22 in our GitHub CI, this is failing the verify-codegen workflow, which executes this go generate command when calling the `make generate` target. See https://github.com/vmware-tanzu/vm-operator/actions/runs/9103082015/job/25024152057?pr=522
[9:11](https://vmware.slack.com/archives/D02H2M35LFM/p1715868672333429)
I also tried upgrading both VMOP and API modules to 1.22 (matching the CI version) but it didn’t help either.

**Review by akutz — APPROVED** [review]
Hi @dougm ,

Thanks for this PR, lgtm! You just need to rebase it to deal with the conflict in the Makefile.

File: Makefile
Comment (dougm) [reply Makefile]: Nice, done

**Pull request merged**
